src/qibolab/result.py

```diff
@@ -83,7 +83,8 @@ def phase(self):
     @cached_property
     def ground_state_probability(self):
         """Computes ground state probability"""
-        return 1 - np.mean(self.shots)
+        # return 1 - np.mean(self.shots)
```

Comment:
If you use the one below, drop the comment.

At this point, an explanation or a reference in the docstring would be appreciated :)

Reply:
I found it easier to just use the array instead of shots on the ExecutionResults so that I could always dump on the array the data on both cases when I'm acquiring in SINGLESHOT and INTEGRATION modes. Also, shots takes an array of integers, but if you do the average on the device of these shot it will return a single float which was not compatible. I didn't want to change it because having both shots and array is useful for the devices that can return i,q values and shots at the same time. I only return one or the other. So we wither change shots or leave this in a way it does not mess with the others

Reply:
I suppose these will work with everyone

```suggestion
        # if self.shots == None:
            return 1 - np.sqrt(self.i**2 + self.q**2)
        else:
           return 1 - np.mean(self.shots)
```

Reply:
Consider that this file and class is also being reworked by @andrea-pasquale in #346, so keep an eye on that :)

However, I'd say that your proposal would be good enough, simply:
- briefly explain the motivation in the docstring
- here a few style suggestions

```py
        if self.shots is None:
            return 1 - np.sqrt(self.measurement)
        
        return 1 - np.mean(self.shots)
```
Namely:
1. use `is` when you compare with `None` (it's a language-level singleton)
2. use `self.measurement`, not to repeat over and over that formula
3. when you `return` in the `if` branch, you do not need an `else` branch, since if you enter the `if` you will break the flow

1. and 3. are typical PyLint suggestions, if you can keep an eye also on what the editor suggests ;) (in case you use VS code with Python extension, or any other Python aware editor)

Reply:
Thanks, for now I spoke with Andrea and I will wait until he merges it to change anything and keep using this for now for the calibrations I need to run.